### ISSUE TITLE
openapi.json yields "Object of type PetSchema is not JSON serializable"

### ISSUE THREAD

**Issue by padhia** [marshmallow-code/flask-smorest#11]
Hello,

I am just starting out with building REST Api in general, so not sure if I am missing something while using this library.

I used the example from the documentation; but when I try to access the `openapi.json`, I get the error mentioned in the description. Is there anything missing with the following code?

```python
@api.definition('Pet')
class PetSchema(ma.Schema):

    class Meta:
        strict = True
        ordered = True

    id = ma.fields.Int(dump_only=True)
    name = ma.fields.String()
```

Also, if you could provide a complete working example, including a working openapi-ui endpoint, that'd be really helpful.

Thanks

**Comment by lafrech** [comment]
The error message seems to indicate that the marshmallow schema was not resolved to json. This shouldn't happen, since the `MarshmallowPlugin`, which takes care of this, is registered by default.

I can't help with the code you're showing. Can you provide the rest of the code, or at least a complete use case?

> Also, if you could provide a complete working example, including a working openapi-ui endpoint, that'd be really helpful.

Yes, I should do that.

**Comment by padhia** [comment]
> The error message seems to indicate that the marshmallow schema was not resolved to json. This shouldn't happen, since the `MarshmallowPlugin`, which takes care of this, is registered by default.
> 

This issue was resolved once I upgraded to the latest git version from 0.7 (pypi). I have a bunch of other REST frameworks installed just to play with, so not sure if that messed up the previous installation.

> Yes, I should do that.

Thanks. It took me a while to figure out what config parameters to set and how to get the openapi UI working. I am sure having a working example will help newbies get a feel of the framework quickly.
